server/src/main/java/org/elasticsearch/search/sort/SortValue.java

```diff
@@ -74,16 +80,7 @@ private SortValue() {
 
     @Override
     public final int compareTo(SortValue other) {
-        /*
-         * It might make sense to try and compare doubles to longs
-         * *carefully* to get a real sort. but it might not. For now
-         * we sort all doubles before all longs.
-         */
-        int typeCompare = getWriteableName().compareTo(other.getWriteableName());
-        if (typeCompare != 0) {
-            return typeCompare;
-        }
-        return compareToSameType(other);
+        return VALUE_COMPARATOR.compare(this, other);
```

Comment:
I think I'd just do the comparisons directly here. I'm not sure what the comparators buys.

Reply:
I wanted to avoid creation of the comparator every time a comparison is made.